fluent-package/yum/fluent-package.spec.in

```diff
@@ -139,29 +138,31 @@ mkdir -p %{buildroot}%{_sysconfdir}/@PACKAGE_DIR@/plugin
 mkdir -p %{buildroot}/tmp/@PACKAGE_DIR@
 
 %pre
-if ! getent group @SERVICE_NAME@ >/dev/null; then
-    /usr/sbin/groupadd -r @SERVICE_NAME@
+if ! getent group @COMPAT_SERVICE_NAME@ >/dev/null; then
+    /usr/sbin/groupadd -r @COMPAT_SERVICE_NAME@
 fi
-if ! getent passwd @SERVICE_NAME@ >/dev/null; then
+if ! getent passwd @COMPAT_SERVICE_NAME@ >/dev/null; then
 %if %{use_suse}
-    /usr/sbin/useradd -r -g @SERVICE_NAME@ -d %{_localstatedir}/lib/@PACKAGE_DIR@ -s /sbin/nologin -c '@SERVICE_NAME@' @SERVICE_NAME@
+    /usr/sbin/useradd -r -g @COMPAT_SERVICE_NAME@ -d %{_localstatedir}/lib/@PACKAGE_DIR@ -s /sbin/nologin -c '@COMPAT_SERVICE_NAME@' @COMPAT_SERVICE_NAME@
 %else
-    /usr/sbin/adduser -r -g @SERVICE_NAME@ -d %{_localstatedir}/lib/@PACKAGE_DIR@ -s /sbin/nologin -c '@SERVICE_NAME@' @SERVICE_NAME@
+    /usr/sbin/adduser -r -g @COMPAT_SERVICE_NAME@ -d %{_localstatedir}/lib/@PACKAGE_DIR@ -s /sbin/nologin -c '@COMPAT_SERVICE_NAME@' @COMPAT_SERVICE_NAME@
 %endif
 fi
 
 %preun
-if [ $1 -eq 0 ]; then
-  %{control_service stop @SERVICE_NAME@}
-  %{disable_service disable @SERVICE_NAME@}
-fi
+%systemd_preun @SERVICE_NAME@.service
 
 %post
+%systemd_post @SERVICE_NAME@.service
 if [ $1 -eq 1 ]; then
-  %{enable_service enable @SERVICE_NAME@}
-fi
-if [ $1 -eq 2 ]; then
-  %{control_service condrestart @SERVICE_NAME@}
+    systemctl is-active @COMPAT_SERVICE_NAME@
+    if [ $? -eq 0 ]; then
+	# When upgrading from v4, td-agent.service will be removed
+	# with %postun scriptlet. fluentd service also inactive even though
+	# td-agent.service is running before upgrade process. Try to
+	# keep running fluentd service, explicitly restart it.
+	systemctl restart @SERVICE_NAME@.service
+    fi
```

Comment:
Is it also workable for upgrading to upper version of fluent-package?
(I'm not sure, will try from now.)

Reply:
I think so.

* upgrade from v4 => td-agent inactive, fluentd active
* upgrade from 5.0.0 (td-agent inactive) => td-agent inactive, fluentd active
* upgrade from 5.0.0 (td-agent active) => td-agent active, fluentd active

Reply:
I've created v5.0.1 and test it.
It also works fine. Thanks!